sherpa/models/model.py

```diff
@@ -46,12 +46,12 @@ def cache_model(cls, pars, xlo, *args, **kwargs):
         digest = ''
         if use_caching:
 
-            data = [ numpy.array(pars).tostring(), str(kwargs.get('integrate',0)), 
+            data = [ numpy.array(pars).tostring(), kwargs.get('integrate', b'0'),
```

Comment:
I think the current code is okay, since it has the str() around the get call, but it isn't ideal: it really should be mapping all inputs to a "sensible" serialization scheme (e.g. 0 or 1). Or at least, that's how I read it. I know I have not really tried to use the caching code, so it wouldn't surprise me if there's some odd corner cases and not enough testing.